### PR TITLE
rodeos filter current_time fix 📦

### PR DESCRIPTION
**Pull request by heifner**
## Change Description

- Fix rodeos kv_table iterator--/++.
- Fix rodeos `current_time` for filters to return last block instead of first block time.

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [x] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: libraries/rodeos/include/eosio/key_value.hpp [hunk 1]
```diff
@@ -268,7 +268,8 @@ class kv_table {
 
       iterator(iterator&& other) :
          itr(std::exchange(other.itr, 0)),
-         itr_stat(std::move(other.itr_stat))
+         itr_stat(std::move(other.itr_stat)),
+         index(std::move(other.index))
```
Comment (huangminghuang) [inline libraries/rodeos/include/eosio/key_value.hpp]: `index` is a const pointer, there's no point to do std::move()

Comment (heifner) [reply libraries/rodeos/include/eosio/key_value.hpp]: yeah, I know. But I was matching cdt code since they are the same.

Comment (heifner) [reply libraries/rodeos/include/eosio/key_value.hpp]: I plan on creating an issue to get kv_table in rodeos to match impl in cdt.

**Review by huangminghuang — APPROVED** [review]

File: libraries/rodeos/include/eosio/key_value.hpp (hunk 1 above)
Comment (heifner) [reply libraries/rodeos/include/eosio/key_value.hpp]: Removed moves that don't move

**Pull request merged**